test/tests/postprocessors/residual/tests

```diff
@@ -0,0 +1,17 @@
+[Tests]
+  design = 'Residual.md'
+  issues = '#28765'
+
+  [test]
+    type = CSVDiff
+    input = 'current_residual.i'
+    csvdiff = 'current_residual_out.csv'
+    # The tolerance needs to be very loose since the gold is a residual norm,
+    # which is prone to significant variations with different solve/execution
+    # types, like serial vs. parallel. We really just want to check that it is
+    # getting the correct iteration's residual, which vary by 6 order of
+    # magnitude here.
```

Comment:
Hmm perhaps I should have read this comment first before writing my above comment. Everything is probably fine then

Reply:
Although I actually would not be surprised if you had a solver residual that differed by 100% so maybe everything is not fine

Reply:
If we are worried here, I could see a custom test executioner / problem that has a pre defined residual that changes pre and post nonlinear convergence checks (on purpose)

Reply:
@joshuahansel I'll say that we can keep this test as-is for now if you don't want to go through the hassle of making something more robust like @GiudGiud suggested. But at some point we'll need to make it more robust (likely)

Reply:
I'll keep as-is. Maybe I should leave a note "call Josh if this test fails"